### ISSUE TITLE
Experiment planing

### ISSUE THREAD

**Issue by 20001LastOrder** [Antolin1/code-inter-dataset-duplication#2]
## Pretraining
- [x] base (12 layers) encoder
    * biased: https://huggingface.co/antolin/csn-small-biased-random-20
    * unbiased: https://huggingface.co/antolin/csn-small-unbiased-random-20

- [x] base encoder + decoder (biased, unbiased) @Antolin1 
    * biased: https://huggingface.co/antolin/csn-small-biased-random-encoder-decoder-20
    * unbiased: https://huggingface.co/antolin/csn-small-unbiased-random-encoder-decoder-20

### Seed
* Biased: 1-10
* Unbiased: 11-20

### Epochs:
* Encoder: 10 
* Encoder + Decoder: **TODO**

### base encoder (10 seeds)
* Biased @20001LastOrder 
    - [x] #3
    - [x] Lora
    - [x] Prefix
* Unbiased @MootezSaaD 
     - [x] FF
    - [x] Lora
    - [x] Prefix

### base encoder + decoder (Python150, CodeTrans, TLC) (10 seeds)
* CodeTrans
  - [x] FF biased
  - [x] FF unbiased
  - [x] Lora biased
  - [x] Lora unbiased
  - [x] Prefix biased
  - [x] Prefix unbiased
 * TLC
  - [x] FF biased
  - [x] FF unbiased
  - [x] Lora biased
  - [x] Lora unbiased
  - [x] Prefix biased
  - [x] Prefix unbiased
 * Python-150
  - [x] FF biased
  - [x] FF unbiased
  - [x] Lora biased
  - [x] Lora unbiased
  - [x] Prefix biased
  - [x] Prefix unbiased

## Layerwise Finetuning (Frozen layers 1 - 11 for large models (encoder))
* 5 seeds (1-5 biased, 11-15 unbiased)
* biased 
* unbiased
